devicetypes/smartthings/zwave-alarm-sound-detector.src/zwave-alarm-sound-detector.groovy

```diff
@@ -0,0 +1,141 @@
+    		capability "Sensor"
+		capability "Battery"
+		capability "Health Check"
+
+		attribute "alarmState", "string"
+
+		fingerprint mfr:"014A", prod:"0005", model:"000F", deviceJoinName: "Ecolink Firefighter"
+	}
+
+	tiles (scale: 2){
+		multiAttributeTile(name:"smoke", type: "lighting", width: 6, height: 4){
+			tileAttribute ("device.alarmState", key: "PRIMARY_CONTROL") {
+				attributeState("clear", label:"clear", icon:"st.alarm.smoke.clear", backgroundColor:"#ffffff")
+				attributeState("smoke", label:"SMOKE", icon:"st.alarm.smoke.smoke", backgroundColor:"#e86d13")
+				attributeState("carbonMonoxide", label:"MONOXIDE", icon:"st.alarm.carbon-monoxide.carbon-monoxide", backgroundColor:"#e86d13")
+			}
+		}
+		valueTile("battery", "device.battery", inactiveLabel: false, decoration: "flat", width: 2, height: 2) {
+			state "battery", label:'${currentValue}% battery', unit:""
+		}
+
+		main "smoke"
+		details(["smoke", "battery"])
+	}
+}
+
+def installed() {
+	sendEvent(name: "checkInterval", value: 2 * 60 * 60 + 2 * 60, displayed: false, data: [protocol: "zwave", hubHardwareId: device.hub.hardwareID])
+	def cmds = []
+	createSmokeOrCOEvents("allClear", cmds)
+	cmds.each { cmd -> sendEvent(cmd) }
+}
+
+def updated() {
+	sendEvent(name: "checkInterval", value: 2 * 60 * 60 + 2 * 60, displayed: false, data: [protocol: "zwave", hubHardwareId: device.hub.hardwareID])
+}
+
+def parse(String description) {
+	def results = []
+	if (description.startsWith("Err")) {
+	    results << createEvent(descriptionText:description, displayed:true)
+	} else {
+		def cmd = zwave.parse(description, [ 0x80: 1, 0x84: 1, 0x71: 2, 0x72: 1 ])
+		if (cmd) {
+			zwaveEvent(cmd, results)
+		}
+	}
+	log.debug "'$description' parsed to ${results.inspect()}"
+	return results
+}
+
+def createSmokeOrCOEvents(name, results) {
+	def text = null
+	switch (name) {
+		case "smoke":
+			text = "$device.displayName smoke was detected!"
+			results << createEvent(name: "smoke",          value: "detected", descriptionText: text, displayed: false)
+			break
+		case "carbonMonoxide":
+			text = "$device.displayName carbon monoxide was detected!"
+			results << createEvent(name: "carbonMonoxide", value: "detected", descriptionText: text, displayed: false)
+			break
+		case "smokeClear":
+			text = "$device.displayName smoke is clear"
+			results << createEvent(name: "smoke",          value: "clear", descriptionText: text, displayed: false)
+			name = "clear"
+			break
+		case "carbonMonoxideClear":
+			text = "$device.displayName carbon monoxide is clear"
+			results << createEvent(name: "carbonMonoxide", value: "clear", descriptionText: text, displayed: false)
+			name = "clear"
+			break
+		case "allClear":
+			text = "$device.displayName all clear"
+			results << createEvent(name: "smoke",          value: "clear", descriptionText: text, displayed: false)
+			results << createEvent(name: "carbonMonoxide", value: "clear", displayed: false)
+			name = "clear"
+			break
```

Comment:
I would not skip default here. You never know - someday someone may modify or copy your code without reading deep enough to understand it.